### PR TITLE
Move mpa.deploy to otx.algorithms.common

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

Move mpa.deploy to otx.algorithms.common.adapters.mmdeploy

### Checklist

- [X] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [X] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [X] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by harimkang** [comment]
@cih9088 Hi, I'm currently working on moving the files from `mpa.module` to `otx`. I'm getting the following unit-test error while moving the deploy related codes (tests/unit/algorithms/common/adapters/mmdeploy/test_deploy_apis.py::TestMMdeployExporter::test_partition), could you please see if I'm missing something? (Some of the modified code is to fix pylint errors, please let me know if you find anything wrong. I'll fix it right away.)
```
    
        with tempfile.TemporaryDirectory() as tempdir:
>           MMdeployExporter.export2openvino(
                tempdir,
                build_classifier,
                config,
                deploy_config,
            )

tests/unit/algorithms/common/adapters/mmdeploy/test_deploy_apis.py:209: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
otx/algorithms/common/adapters/mmdeploy/apis.py:244: in export2openvino
    MMdeployExporter.onnx2openvino(
otx/algorithms/common/adapters/mmdeploy/apis.py:330: in onnx2openvino
    raise e
otx/algorithms/common/adapters/mmdeploy/apis.py:317: in onnx2openvino
    openvino_api.from_onnx(onnx_ready_path, output_dir, input_info, output_names, mo_options)
venv/lib/python3.9/site-packages/mmdeploy/apis/core/pipeline_manager.py:356: in _wrap
    return self.call_function(func_name_, *args, **kwargs)
venv/lib/python3.9/site-packages/mmdeploy/apis/core/pipeline_manager.py:326: in call_function
    return self.call_function_local(func_name, *args, **kwargs)
venv/lib/python3.9/site-packages/mmdeploy/apis/core/pipeline_manager.py:275: in call_function_local
    return pipe_caller(*args, **kwargs)
venv/lib/python3.9/site-packages/mmdeploy/apis/core/pipeline_manager.py:107: in __call__
    ret = func(*args, **kwargs)
venv/lib/python3.9/site-packages/mmdeploy/backend/openvino/onnx2openvino.py:114: in from_onnx
    mo_output = run(command, stdout=PIPE, stderr=PIPE, shell=True, check=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

input = None, capture_output = False, timeout = None, check = True
popenargs = ('mo --input_model="/tmp/tmp81n5d140/partition_ready.onnx" --output_dir="/tmp/tmp81n5d140" --output="" --input="input" --input_shape="[1, 3, 50, 50]" --model_name "partition" ',), kwargs = {'shell': True, 'stderr': -1, 'stdout': -1}
process = <Popen: returncode: 1 args: 'mo --input_model="/tmp/tmp81n5d140/partition_re...>, stdout = b'', stderr = b'[ ERROR ]  No node with name input\n', retcode = 1

    def run(*popenargs,
            input=None, capture_output=False, timeout=None, check=False, **kwargs):
        """Run command with arguments and return a CompletedProcess instance.
    
        The returned instance will have attributes args, returncode, stdout and
        stderr. By default, stdout and stderr are not captured, and those attributes
        will be None. Pass stdout=PIPE and/or stderr=PIPE in order to capture them.
    
        If check is True and the exit code was non-zero, it raises a
        CalledProcessError. The CalledProcessError object will have the return code
        in the returncode attribute, and output & stderr attributes if those streams
        were captured.
    
        If timeout is given, and the process takes too long, a TimeoutExpired
        exception will be raised.
    
        There is an optional argument "input", allowing you to
        pass bytes or a string to the subprocess's stdin.  If you use this argument
        you may not also use the Popen constructor's "stdin" argument, as
        it will be used internally.
    
        By default, all communication is in bytes, and therefore any "input" should
        be bytes, and the stdout and stderr will be bytes. If in text mode, any
        "input" should be a string, and stdout and stderr will be strings decoded
        according to locale encoding, or by "encoding" if set. Text mode is
        triggered by setting any of text, encoding, errors or universal_newlines.
    
        The other arguments are the same as for the Popen constructor.
        """
        if input is not None:
            if kwargs.get('stdin') is not None:
                raise ValueError('stdin and input arguments may not both be used.')
            kwargs['stdin'] = PIPE
    
        if capture_output:
            if kwargs.get('stdout') is not None or kwargs.get('stderr') is not None:
                raise ValueError('stdout and stderr arguments may not be used '
                                 'with capture_output.')
            kwargs['stdout'] = PIPE
            kwargs['stderr'] = PIPE
    
        with Popen(*popenargs, **kwargs) as process:
            try:
                stdout, stderr = process.communicate(input, timeout=timeout)
            except TimeoutExpired as exc:
                process.kill()
                if _mswindows:
                    # Windows accumulates the output in a single blocking
                    # read() call run on child threads, with the timeout
                    # being done in a join() on those threads.  communicate()
                    # _after_ kill() is required to collect that and add it
                    # to the exception.
                    exc.stdout, exc.stderr = process.communicate()
                else:
                    # POSIX _communicate already populated the output so
                    # far into the TimeoutExpired exception.
                    process.wait()
                raise
            except:  # Including KeyboardInterrupt, communicate handled that.
                process.kill()
                # We don't call process.wait() as .__exit__ does that for us.
                raise
            retcode = process.poll()
            if check and retcode:
>               raise CalledProcessError(retcode, process.args,
                                         output=stdout, stderr=stderr)
E               subprocess.CalledProcessError: Command 'mo --input_model="/tmp/tmp81n5d140/partition_ready.onnx" --output_dir="/tmp/tmp81n5d140" --output="" --input="input" --input_shape="[1, 3, 50, 50]" --model_name "partition" ' returned non-zero exit status 1.

../../../anaconda3/lib/python3.9/subprocess.py:528: CalledProcessError
============================================================================================================= short test summary info ==============================================================================================================
FAILED tests/unit/algorithms/common/adapters/mmdeploy/test_deploy_apis.py::TestMMdeployExporter::test_partition - subprocess.CalledProcessError: Command 'mo --input_model="/tmp/tmp81n5d140/partition_ready.onnx" --output_dir="/tmp/tmp81n5d140" --output="" --input="input" --input_shape="[1, 3, 50, 50]" --model_name "partition" ' returned non-zero exit s...
========================================================================================================== 1 failed, 12 warnings in 6.18s ==========================================================================================================
```

**Comment by cih9088** [comment]
>(tests/unit/algorithms/common/adapters/mmdeploy/test_deploy_apis.py::TestMMdeployExporter::test_partition), could you please see if I'm missing something? (Some of the modified code is to fix pylint errors, please let me know if you find anything wrong. I'll fix it right away.)

@harimkang  Fixed.

File: otx/mpa/det/exporter.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def naive_export(output_dir, model_builder, precision, cfg, model_name="model"):
         from mmdet.apis.inference import LoadImage
         from mmdet.datasets.pipelines import Compose
 
-        from ..deploy.apis import NaiveExporter
+        from ...algorithms.common.adapters.mmdeploy.apis import NaiveExporter
```
Comment (jaegukhyun) [inline otx/mpa/det/exporter.py]:
```suggestion
        from otx.algorithms.common.adapters.mmdeploy.apis import NaiveExporter
```

File: otx/mpa/exporter_mixin.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def mmdeploy_export(
         deploy_cfg,
         model_name="model",
     ):
-        from .deploy.apis import MMdeployExporter
+        from ..algorithms.common.adapters.mmdeploy.apis import MMdeployExporter
```
Comment (jaegukhyun) [inline otx/mpa/exporter_mixin.py]:
```suggestion
        from otx.algorithms.common.adapters.mmdeploy.apis import MMdeployExporter
```

File: otx/mpa/seg/exporter.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def naive_export(output_dir, model_builder, precision, cfg, model_name="model"):
         from mmseg.apis.inference import LoadImage
         from mmseg.datasets.pipelines import Compose
 
-        from ..deploy.apis import NaiveExporter
+        from ...algorithms.common.adapters.mmdeploy.apis import NaiveExporter
```
Comment (jaegukhyun) [inline otx/mpa/seg/exporter.py]:
```suggestion
        from otx.algorithms.common.adapters.mmdeploy.apis import NaiveExporter
```

File: otx/mpa/cls/exporter.py [hunk 4]
```diff
@@ -47,7 +47,7 @@ def model_builder_helper(*args, **kwargs):
     def naive_export(output_dir, model_builder, precision, cfg, model_name="model"):
         from mmcls.datasets.pipelines import Compose
 
-        from ..deploy.apis import NaiveExporter
+        from ...algorithms.common.adapters.mmdeploy.apis import NaiveExporter
```
Comment (jaegukhyun) [inline otx/mpa/cls/exporter.py]:
```suggestion
        from otx.algorithms.common.adapters.mmdeploy.apis import NaiveExporter
```

File: otx/algorithms/common/adapters/mmdeploy/utils/mmdeploy.py [hunk 5]
```diff
@@ -31,12 +42,14 @@ def mmdeploy_init_model_helper(ctx, model_checkpoint=None, cfg_options=None, **k
...
-def update_deploy_cfg(onnx_path, deploy_cfg, mo_options={}):
+def update_deploy_cfg(onnx_path, deploy_cfg, mo_options=None):
+    """Update the 'deploy_cfg' configuration file based on the ONNX model specified by 'onnx_path'."""
+
     from mmdeploy.utils import get_backend_config, get_ir_config
 
     onnx_model = onnx.load(onnx_path)
     ir_config = get_ir_config(deploy_cfg)
-    backend_config = get_backend_config(deploy_cfg)
+    _ = get_backend_config(deploy_cfg)
```
Comment (jaegukhyun) [inline otx/algorithms/common/adapters/mmdeploy/utils/mmdeploy.py]:
```suggestion
    get_backend_config(deploy_cfg)
```

**Review by jaegukhyun — COMMENTED** [review]
I left some minor fix suggest, If you don't want to change, then just resolve them.

File: otx/algorithms/common/adapters/mmdeploy/utils/operations_domain.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+"""Add domina function."""
```
Comment (JihwanEom) [inline otx/algorithms/common/adapters/mmdeploy/utils/operations_domain.py]:
```suggestion
"""Add domain function."""
```

**Comment by harimkang** [comment]
@jaegukhyun @JihwanEom I think we can merge this, but can you double check?

**Review by sungmanc — APPROVED** [review]
Looks nice to me, it seems that you resolves plenty of pre-commit issues

**Review by JihwanEom — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**